extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/runtime/JpaOperations.java

```diff
@@ -263,7 +279,12 @@ public static Object findById(Class<?> entityClass, Object id, LockModeType lock
         String findQuery = createFindQuery(entityClass, query, paramCount(params));
         EntityManager em = getEntityManager();
         // FIXME: check for duplicate ORDER BY clause?
-        Query jpaQuery = em.createQuery(sort != null ? findQuery + toOrderBy(sort) : findQuery);
+        Query jpaQuery;
+        if (isNamedQuery(query)) {
```

Comment:
And here.

Reply:
same

Reply:
Same here, it comes straight from the user. Now, I'll admit it looks like we're missing a null check too…

Reply:
same, fixed